app/src/sharedTest/java/org/oppia/android/app/devoptions/platformparameters/PlatformParametersFragmentTest.kt

```diff
@@ -1593,51 +1628,147 @@ class PlatformParametersFragmentTest {
   }
 
   @Test
-  fun testPlatformParametersFragment_removeTextFromParameterAndFocusChange_doesNotShowErrorMsg() {
+  fun testPlatformParametersFragment_modifyParam_navigateBack_clickSave_showRestartDialogExitApp() {
     setUpTestApplicationComponent()
-    launch(PlatformParametersTestActivity::class.java).use {
-      it.onActivity { activity ->
+    val exception = assertThrows<SecurityException>() {
+      launch(PlatformParametersTestActivity::class.java).use {
         testCoroutineDispatchers.runCurrent()
-        val recyclerView =
-          activity.findViewById<RecyclerView>(R.id.platform_parameters_recycler_view)
-        val viewHolder1 = recyclerView.findViewHolderForAdapterPosition(1)
-        val editText1 =
-          viewHolder1?.itemView?.findViewById<EditText>(R.id.platform_parameter_input_edit_text)
 
+        scrollToPosition(0)
         onView(
           atPositionOnView(
             recyclerViewId = R.id.platform_parameters_recycler_view,
-            position = 1,
-            targetViewId = R.id.platform_parameter_input_edit_text
+            position = 0,
+            targetViewId = R.id.platform_parameter_switch
           )
-        ).perform(editTextInputAction.replaceText(""))
+        ).perform(click())
+
+        pressBack()
         testCoroutineDispatchers.runCurrent()
 
-        val viewHolder2 = recyclerView.findViewHolderForAdapterPosition(2)
-        val editText2 =
-          viewHolder2?.itemView?.findViewById<EditText>(R.id.platform_parameter_input_edit_text)
+        onView(withText(R.string.pending_changes_dialog_save_button_text))
+          .inRoot(isDialog())
+          .perform(click())
 
-        editText1?.performAccessibilityAction(AccessibilityNodeInfo.ACTION_FOCUS, Bundle())
-        editText2?.performAccessibilityAction(AccessibilityNodeInfo.ACTION_FOCUS, Bundle())
+        testCoroutineDispatchers.runCurrent()
 
-        scrollToPosition(1)
+        onView(withText(R.string.app_restart_dialog_title))
+          .inRoot(isDialog())
+          .check(matches(isDisplayed()))
+      }
```

Comment:
Here and the below similar test case + similar cases in feature flags as well. 

Perform restart to actually test the system exit not just via the indirect function call. 
Also, suggestion for the test name for these cases:

```kt
fun testPlatformParametersFragment_saveChangesOnBackNavigation_showsRestartDialogAndExitsApp()
```

```suggestion
        onView(withText(R.string.app_restart_dialog_title))
          .inRoot(isDialog())
          .check(matches(isDisplayed()))
          .perform(click())
      }
```

Reply:
Updated